### ISSUE TITLE
serviceId.ribbon.MaxAutoRetriesNextServer not working in zuul

### ISSUE THREAD

**Issue by lishengsam** [spring-cloud/spring-cloud-netflix#2810]
I want to retry the request that was failed in zuul server, such connection refused, read timeout, hystrix timeout.
But I set the option of serviceId.ribbon.MaxAutoRetriesNextServer=2 is not working in zuul When the GET/POST request was connection refused and read timeout

My config of zuul server as follows:
1. pom.xml
```
  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>org.springframework.cloud</groupId>
        <artifactId>spring-cloud-dependencies</artifactId>
        <version>**Edgware.RELEASE**</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
    </dependencies>
  </dependencyManagement>

<dependencies>
    <dependency>
      <groupId>org.springframework.cloud</groupId>
      <artifactId>spring-cloud-starter-zipkin</artifactId>
    </dependency>
    <dependency>
      <groupId>org.springframework.cloud</groupId>
      <artifactId>spring-cloud-starter-eureka</artifactId>
    </dependency>
    <dependency>
      <groupId>org.springframework.cloud</groupId>
      <artifactId>spring-cloud-starter-zuul</artifactId>
    </dependency>
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-devtools</artifactId>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>org.springframework.retry</groupId>
      <artifactId>spring-retry</artifactId>
    </dependency>
  </dependencies>
```
2. application.yaml
```
spring:
  application:
    name: zuul-server

zuul:
  routes:
    push-token-route:
      path: /token/**
      serviceId : push-token
      stripPrefix: false  
  semaphore:
    max-semaphores: 1000
  ribbon:
    eager-load:
      enabled: true
  retryable: true
  ribbon-isolation-strategy: semaphore

hystrix:
  command:
    push-token:
      execution:
        isolation:
          strategy: SEMAPHORE
          semaphore:
            maxConcurrentRequests: 30
          thread:
            timeoutInMilliseconds: 2000

push-token:
  ribbon:
    MaxAutoRetriesNextServer: 2
    MaxAutoRetries: 0
    OkToRetryOnAllOperations: true
    ConnectTimeout: 100
    ReadTimeout: 400

management:
  security:
    enabled: false
```

Is my config wrong?

**Comment by ryanjbaxter** [comment]
What is the status code returned?  Is there any kind of exception in the logs?

**Comment by lishengsam** [comment]
sorry, I had found that it had worked，but I did not noticed.  zuul fetch the push-token serivce list from eureka server, which have 4 instances info but 2 intances info is dirty(they was down, but not deleted due to the self prevention). This situation confused me, which is that when first request hit the one of 2 dirty instances, zuul didn't print any log about the connection refused and retry the the another dirty instances, failed again and retry the normal instance that had a beakpoint for timeout , so the zuul print the read timeout exception and return 500 status. Zuul only print the last retry request exception log ，so I think it was the first request and not retry. When I set MaxAutoRetriesNextServer=3, each request was success.
